src/main/anovos/data_transformer/transformers.py

```diff
@@ -3422,6 +3421,8 @@ def outlier_categories(
     if output_mode not in ("replace", "append"):
         raise TypeError("Invalid input for output_mode")
 
+    idf = idf.persist(pyspark.StorageLevel.MEMORY_AND_DISK)
```

Comment:
Have you tested the runtime with and without `persist`? By default, `idf` when being read in, is cached inside Spark Session memory already. This might cause some potential runtime/memory issue. Also, if you decide to put `persist`, you should `unpersist` it at the end of the function as well.

Reply:
From our experience using Spark Scala - when dataset is huge and also read multiple times in the code then we should use persist (memory and disk if its very huge, only memory if its not big enough). We found significant performance improvement in our spark scala code in the past after using it. Here also we used 100 GB dataset and were using data frame multiple times for each column so persist is useful. I will test with and without persist and come back with results. I will also use unpersist in the end.